server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -1168,6 +1168,13 @@ private void parseSource(DefaultSearchContext context, SearchSourceBuilder sourc
         SearchExecutionContext searchExecutionContext = context.getSearchExecutionContext();
         context.from(source.from());
         context.size(source.size());
+        if (source.rank() != null) {
```

Comment:
we need to validate that there aren't any `sorts` provided and throw if there are.

Reply:
Allowing sort actually doesn't break anything. But the behavior is certainly odd as it will apply to all queries. So I guess this should be part of the restrictions list.

Reply:
I restricted a number of things including sort. Since we do part of the ranking on the shard, we no longer pass FieldDoc to the coordinator, so we need to re-think about how to appropriately support sort. I think we may actually be able to replace RankDoc with FieldDoc in the future, but I think that is best saved for a follow up.